static/js/metrics-cardinality.js

```diff
@@ -0,0 +1,178 @@
+    new agGrid.Grid(document.getElementById('tagKeysValuesGrid'), tagKeysValuesGridOptions);
+
+    function fetchData(startEpoch, endEpoch) {
+        // Fetch total unique series
+        fetch('http://localhost:5122/metrics-explorer/api/v1/series-cardinality', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ startEpoch, endEpoch })
+        })
+            .then(response => response.json())
+            .then(data => {
+                totalUniqueSeries.textContent = data.seriesCardinality;
+            });
+
+        // Fetch tag keys with most series
+        fetch('http://localhost:5122/metrics-explorer/api/v1/tag-keys-with-most-series', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ startEpoch, endEpoch, limit: 10 })
+        })
+            .then(response => response.json())
+            .then(data => {
+                tagKeysGridOptions.api.setRowData(data.tagKeys);
+            });
+
+        // Fetch tag pairs with most series
+        fetch('http://localhost:5122/metrics-explorer/api/v1/tag-pairs-with-most-series', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ startEpoch, endEpoch, limit: 10 })
+        })
+            .then(response => response.json())
+            .then(data => {
+                tagPairsGridOptions.api.setRowData(data.tagPairs);
+            });
+
+        // Fetch tag keys with most unique values
+        fetch('http://localhost:5122/metrics-explorer/api/v1/tag-keys-with-most-values', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ startEpoch, endEpoch, limit: 10 })
+        })
+            .then(response => response.json())
+            .then(data => {
+                tagKeysValuesGridOptions.api.setRowData(data.tagKeys);
+            });
+    }
+
+    function updateSelectedTimeRange(element) {
+        document.querySelectorAll('.range-item').forEach(item => {
+            item.classList.remove('active');
+        });
+        element.classList.add('active');
+        selectedTimeRange.textContent = element.textContent;
+    }
+
+    document.querySelectorAll('.range-item').forEach(item => {
+        item.addEventListener('click', function () {
+            var startEpoch = this.id;
+            var endEpoch = 'now';
+            fetchData(startEpoch, endEpoch);
+            updateSelectedTimeRange(this);
+        });
+    });
+
+    document.getElementById('customrange-btn').addEventListener('click', function () {
+        var startEpoch = document.getElementById('date-start').value + 'T' + document.getElementById('time-start').value + ':00Z';
```

Comment:
For a custom time range, you need to send it in epoch form. Please take a look at the other screen to see how it is done.